x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/QueryRulesIndexService.java

```diff
@@ -263,6 +267,8 @@ public void deleteQueryRuleset(String resourceName, ActionListener<DeleteRespons
                     l.onResponse(deleteResponse);
                 })
             );
+        } catch (IndexNotFoundException e) {
```

Comment:
I'm curious - why didn't the `DelegatingIndexNotFoundActionListener` above catch this? 🤔

Reply:
I think you caught something not working, moving back to draft

Reply:
OK, I updated this.